src/awsexplorer/activation.ts

```diff
@@ -70,62 +61,46 @@ export async function activate(args: {
     updateAwsExplorerWhenAwsContextCredentialsChange(awsExplorer, args.context.awsContext, globals.context)
 }
 
-async function tryAutoConnect(awsContext: AwsContext) {
-    try {
-        if (!didTryAutoConnect && !(await awsContext.getCredentials())) {
-            getLogger().debug('credentials: attempting autoconnect...')
-            didTryAutoConnect = true
-            const loginManager = new LoginManager(awsContext, new CredentialsStore())
-            await loginWithMostRecentCredentials(new CredentialsSettings(), loginManager)
-        }
-    } catch (err) {
-        getLogger().error('credentials: failed to auto-connect: %O', err)
-        showViewLogsMessage(localize('AWS.credentials.autoconnect.fatal', 'Exception occurred while connecting'))
-    }
-}
-
 async function registerAwsExplorerCommands(
     context: ExtContext,
     awsExplorer: AwsExplorer,
     toolkitOutputChannel: vscode.OutputChannel
 ): Promise<void> {
     context.extensionContext.subscriptions.push(
-        vscode.commands.registerCommand('aws.showRegion', async () => {
+        // Internal-only command to avoid circular reference from the `Commands` module.
+        Commands.register(
+            {
+                id: 'aws.tryAutoConnect',
+                autoconnect: false,
+            },
+            async () => {
+                return await LoginManager.tryAutoConnect(globals.awsContext)
+            }
+        ),
+        Commands.register('aws.showRegion', async () => {
             try {
                 await globals.awsContextCommands.onCommandShowRegion()
             } finally {
                 recordAwsShowRegion()
                 recordVscodeActiveRegions({ value: awsExplorer.getRegionNodesSize() })
             }
-        })
-    )
-
-    context.extensionContext.subscriptions.push(
-        vscode.commands.registerCommand('aws.submitFeedback', async () => {
+        }),
+        Commands.register('aws.submitFeedback', async () => {
             await submitFeedback(context)
-        })
-    )
-
-    context.extensionContext.subscriptions.push(
-        vscode.commands.registerCommand('aws.refreshAwsExplorer', async (passive: boolean = false) => {
+        }),
+        Commands.register('aws.refreshAwsExplorer', async (passive: boolean = false) => {
```

Comment:
We can do this later but this is a place where `declare` is nice. We can move the logic to where it's actually implemented and then consumers can refer directly to the command rather than `aws.refreshAwsExplorer`.